scripts/run-benchmarks.py

```diff
@@ -0,0 +1,200 @@
+import subprocess
+import os
+
+import click
+import cpuinfo
+import requests
+
+BASE_REPO_URL = 'https://github.com/lumol-org/lumol.git'
+
+INDIVIDUAL_BENCH_TEMPLATE = """
+<details><summary>{sha} {title}</summary>
+  <p>\n\n
+\n\n
+```bash\n
+{body}\n\n
+```\n\n
+
+</p></details>
+"""
+
+COMPARISON_TEMPLATE = """
+<details><summary>{sha} {title}</summary>
+  <p>\n\n
+\n\n
+```bash\n
+{body}\n\n
+```\n\n
+
+</p></details>
+"""
+
+
+def _get_environment_variable(name):
+    var = os.environ.get(name, None)
+    if var is None:
+        msg = 'The environment variable {} must be set to access Github API'.format(name)
+        raise Exception(msg)
+
+    return var
+
+
+def check_for_cargo_benchcmp():
+    try:
+        subprocess.check_output('cargo benchcmp --help', shell=True)
+    except subprocess.CalledProcessError as e:
+        raise Exception("cargo-benchcmp is not installed, please install with "
+                        "`cargo install cargo-benchcmp`")
+
+
+def request_api(endpoint, data=None):
+    url = 'https://api.github.com/repos/lumol-org/lumol' + endpoint
+    username = _get_environment_variable('LUMOL_GH_USERNAME')
+    token = _get_environment_variable('LUMOL_GH_TOKEN')
+
+    if data is None:
+        r = requests.get(url, auth=(username, token))
+    else:
+        r = requests.post(url, auth=(username, token), json=data)
+
+    return r.json()
+
+
+def get_master_commit():
+    cmd = 'git log --oneline upstream/master -n 1'
+    out = subprocess.check_output(cmd, shell=True).decode('utf-8')
+    sha, _, title = out.rstrip().partition(' ')
+    return sha, title
+
+
+def get_commit_descriptions(n_commits):
+    """
+    Get hash and title of the `n_commits` latest commits on the branch.
+
+    Also adds the commit at the HEAD of master in the end. If this
+    commit is met earlier, stops at this commit (the master commit
+    is guaranteed to be at the end of the result.
+
+    :param n_commits:
+    :return:
+    """
+    cmd = 'git log --oneline  -n {}'.format(n_commits)
+    out = subprocess.check_output(cmd, shell=True).decode('utf-8')
+    master_sha, master_title = get_master_commit()
+
+    descriptions = []
+    for line in out.split('\n'):
+        if line.rstrip() == '':
+            continue
+        sha, _, title = line.partition(' ')
+        descriptions.append((sha, title))
+        if sha == master_sha:
+            break
+
+    if descriptions[-1][0] != master_sha:
+        descriptions.append((master_sha, master_title))
+
+    return descriptions
+
+
+def clean_repo():
+    subprocess.call('git checkout master', shell=True)
+    subprocess.call('git branch -D _bot_pr', shell=True)
+    subprocess.call('git remote remove _bot_remote', shell=True)
+
+
+def setup_repo(pr_id):
```

Comment:
Shouldn't this be a context manager? Else is can leave the repo in a dirty state if there is an exception somewhere.

As you are already cleaning before anything, this should be fine, but it might be cleaner to ensure the cleaning at the end of the script.